content/documentation/configuration.md

```diff
@@ -169,16 +169,41 @@ This is the default and only API implementation available. It provides a REST AP
-|`headers` | GET * CORS Headers | A key-values map of headers the API endpoint should return with each response. i.e. `"headers": {"header_name": [ "v1", "v2" ] }`. It comes initialized by default to support CORs values so that things like the IPFS GUI can interact with Cluster too.
+|`headers` | `null` | A `key: [values]` map of headers the API endpoint should return with each response to `GET`, `POST`, `DELETE` requests. i.e. `"headers": {"header_name": [ "v1", "v2" ] }`. Do not place CORS headers here, as they are fully handled by the options below. |
+|`cors_allowed_origins`| `["*"]` | CORS Configuration: values for `Access-Control-Allow-Origin`. |
+|`cors_allowed_methods`| `["GET"]` | CORS Configuration: values for `Access-Control-Allow-Methods`. |
+|`cors_allowed_headers`| `[]` | CORS Configuration: values for `Access-Control-Allow-Headers`. |
+|`cors_exposed_headers`| `["Content-Type", "X-Stream-Output", "X-Chunked-Output", "X-Content-Length"]` | CORS Configuration: values for `Access-Control-Expose-Headers`. |
+|`cors_allow_credentials`|  `true` | CORS Configuration: value for `Access-Control-Allow-Credentials`. |
+|`cors_max_age`|  `"0s"` | CORS Configuration: value for `Access-Control-Max-Age`. |
 
 The REST API component automatically, and additionally, exposes the HTTP API as a libp2p service on the main libp2p cluster Host (which listens on port `9096`). Exposing the HTTP API as a libp2p service allows users to benefit from the channel encryption provided by libp2p. Alternatively, the API supports specifying a fully separate libp2p Host by providing `id`, `private_key` and `libp2p_listen_multiaddress`. When using a separate Host, it is not necessary for an API consumer to know the cluster secret. Both the HTTP and the libp2p endpoints are supported by the [API Client](https://godoc.org/github.com/ipfs/ipfs-cluster/api/rest/client) and by [`ipfs-cluster-ctl`](/documentation/ipfs-cluster-ctl/).
 
+##### > `ipfsproxy`
+
+This component provides the IPFS Proxy Endpoint. This is an API which mimics the IPFS daemon. Some requests (pin, unpin, add) are hijacked and handled by Cluster. Others are simply forwarded to the IPFS daemon specified by `node_multiaddress`. The component is by default configured to mimic CORS headers configurations as present in the IPFS daemon. For
+that it triggers accessory requests to them (like CORS preflights).
+
+|Key|Default|Description|
+|:---|:-------|:-----------|
+|`node_multiaddress` | `"/ip4/127.0.0.1/tcp/5001"` | The listen addres of the IPFS daemon API. |
+|`listen_multiaddress` | `"/ip4/127.0.0.1/tcp/9095"` | The proxy endpoint listening address. |
+|`node_https` | `false` | Use HTTPS to talk to the IPFS API endpoint (experimental). |
+|`read_timeout` | `"0s"` | Parameters for https://godoc.org/net/http#Server . Note setting this value might break adding to cluster, if the timeout is shorter than the time it takes to add something to the cluster. |
```

Comment:
0s feels like it will always time out. Does 0s means infinite timeout. I wonder if it is worth using some other default value.

Reply:
This is consistent with Go http server config, and these are passed directly to it.